src/components/MovieDetail/MovieDetailContainer.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import MovieDetail from './MovieDetail';
+import Error from '../Error/Error';
+import fetchRequests from '../../fetchRequests';
+import { trackPromise } from 'react-promise-tracker';;
+
+class MovieDetailContainer extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      selectedDetails: null,
+      error: null,
+    }
+  }
+
+  componentDidMount() {
+   const {id} = this.props.match.params
+    trackPromise(
+      fetchRequests.getSelectedMovie(id)
+      .then(data => this.setState({selectedDetails: data.movie}))
+      .catch(error => {
+        this.setState({error: error})
+      })
+    )
+  }
+
+  render() {
+    return (
+    <>
+      {this.state.error && <Error status={this.state.error.status} text={this.state.error.statusText}/>}
+      {this.state.selectedDetails && <MovieDetail movie={this.state.selectedDetails}/>}
+    </>
+    )
+  }
+}
+
+export default MovieDetailContainer
```

Comment:
So this solution is really cool and our code looks way better. We talked about it a lot before we tried this way. I STILL am trying to think of a way to keep all the data in `App.js` and might suggest a change at some point but idk if I can figure it out. It might be impossible. That `componentDidMount()` is really clutch.